outbox.js

```diff
@@ -0,0 +1,205 @@
+class DistributedOutbox extends HTMLElement {
+  constructor() {
+    super();
+    this.renderedItems = new Set(); // Tracks rendered items
+    this.numPosts = 1; // Default value
+    this.page = 1; // Default value
+    this.totalPages = 0; // Keep track of total pages
+  }
+
+  static get observedAttributes() {
+    return ["url", "num-posts", "page"];
+  }
+
+  connectedCallback() {
+    // Use attributes or default values
+    this.numPosts = parseInt(this.getAttribute("num-posts")) || this.numPosts;
+    this.page = parseInt(this.getAttribute("page")) || this.page;
+    this.loadOutbox(this.getAttribute("url"));
+  }
+
+  async loadOutbox(outboxUrl) {
+    this.clearContent();
+    for await (const item of this.fetchOutboxItems(outboxUrl)) {
+      // Check if the item hasn't been rendered
+      if (!this.renderedItems.has(item.object)) {
+        this.renderItem(item);
+        // Mark as rendered
+        this.renderedItems.add(item.object);
+      }
+    }
+  }
+
+  async *fetchOutboxItems(outboxUrl) {
+    if (!outboxUrl) {
+      console.error("No outbox URL provided");
+      return;
+    }
+
+    try {
+      const response = await fetch(outboxUrl);
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      const outboxData = await response.json();
+
+      this.totalPages = Math.ceil(
+        outboxData.orderedItems.length / this.numPosts
+      );
+      // Prevent page number from going beyond total pages
+      this.page = Math.min(this.page, this.totalPages);
+
+      // Simulate pagination by slicing the items array
+      const startIndex = (this.page - 1) * this.numPosts;
+      const endIndex = startIndex + this.numPosts;
+      const paginatedItems = outboxData.orderedItems.slice(
+        startIndex,
+        endIndex
+      );
+
+      for (const item of paginatedItems) {
+        yield item;
+      }
+    } catch (error) {
+      console.error("Error fetching outbox:", error);
+    }
+  }
+
+  renderItem(item) {
+    const activityElement = document.createElement("distributed-activity");
+    activityElement.setAttribute("type", item.type);
+    activityElement.setAttribute("data", JSON.stringify(item));
```

Comment:
Instead of setting the data as an attribute, maybe we can set it as a regular property? Else maybe the URL of the item?

Reply:
It fixed this
![Screenshot 2024-02-22 at 2 56 56 AM](https://github.com/hyphacoop/reader.distributed.press/assets/68826419/a37e2d8e-fa89-4b0e-a85f-321fd5fee611)